tests/ZonedDateTimeTest.php

```diff
@@ -482,4 +486,435 @@ public function providerCompareTo() : array
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-01-20T12:34:54.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->minusSeconds(2));
+    }
+
+    public function testIsBetweenInclusive()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $fromZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $localDateTime = '2015-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $toZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $localDateTime = '2018-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $notIncluZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertTrue($fromZonedDateTime->isBetweenInclusive($fromZonedDateTime, $toZonedDateTime));
+        $this->assertFalse($fromZonedDateTime->isBetweenInclusive($toZonedDateTime, $notIncluZonedDateTime));
+    }
+
+    public function testIsBetweenExclusive()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $fromZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $localDateTime = '2015-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $toZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $localDateTime = '2014-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $incluZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertTrue($incluZonedDateTime->isBetweenExclusive($fromZonedDateTime, $toZonedDateTime));
+        $this->assertFalse($fromZonedDateTime->isBetweenExclusive($fromZonedDateTime, $toZonedDateTime));
+    }
+
+    public function testIsFuture()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $pastZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertFalse($pastZonedDateTime->isFuture());
+    }
+
+    public function testIsPast()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '1998-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $futureZonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertTrue($futureZonedDateTime->isPast());
```

Comment:
You're implicitly testing with SystemClock (the current time), which is not a good idea in tests.
It would be nice to provide a FixedClock to `isFuture()` and `isPast()`.

Also, it would be nice to have a data provider with a few tests in there: the provider can return [$clockTimestamp, $dateTime, $isFuture], with a couple timestamps, and at least 2 datetimes each (one past, one future).

You can actually use the same data provider for both test methods:
The first one will do `assertSame($isFuture, $zonedDateTime->isFuture())`, while the second one will do `assertSame(! $isFuture, $zonedDateTime->isPast())`.